Users Menu: add new column listing the linked guest author.

jeherve opened the pull request:
Fixes #494

The post count in the Users Menu changes when you've linked an account to a guest author. That is not always clear to everyone, as explained in #494. 

If we add a new column to the menu that shows that guest author, right before the post count, I think it makes things a bit clearer.

**Before**

<img width="1119" alt="screenshot 2018-04-11 at 22 01 36" src="https://user-images.githubusercontent.com/426388/38640255-16cab22a-3dd4-11e8-803e-8d92af1f52d9.png">


**After**

<img width="1126" alt="screenshot 2018-04-11 at 22 00 54" src="https://user-images.githubusercontent.com/426388/38640247-11ea1660-3dd4-11e8-9de8-63dc92134229.png">

What do you think?

**Testing**

To test, go to Users > Guest Authors, create a few guest authors and link them to existing accounts on your site like so:

<img width="340" alt="screenshot 2018-04-11 at 22 06 41" src="https://user-images.githubusercontent.com/426388/38640460-a1b9c402-3dd4-11e8-9fba-743f8ef647b9.png">

Then visit the Users menu and make sure the information and links displayed there are correct.

rebeccahum commented:
@jeherve That's not a bad idea because it would mirror the Guest Author user panel and would be more consistent:

![image](https://user-images.githubusercontent.com/16962021/38645349-fd116ea6-3da0-11e8-93a6-757dd0c86b6b.png)

GaryJones commented:
Only the [last commit](https://github.com/Automattic/Co-Authors-Plus/pull/502/commits/3b8e09f4167cc23f4edc4ffae403c4997cc95001) is relevant.

jeherve commented:
I do not really contributo to Co-Authors Plus anymore, but I rebased that branch if that can make your job easier. The PHPCS error doesn't appear to be related to this PR though:

```
FILE: .../runner/work/Co-Authors-Plus/Co-Authors-Plus/co-authors-plus.php
----------------------------------------------------------------------
FOUND 1 ERROR AFFECTING 1 LINE
----------------------------------------------------------------------
 609 | ERROR | count_user_posts() is highly discouraged due to not
     |       | being cached; please use wpcom_vip_count_user_posts()
     |       | instead.
     |       | (WordPressVIPMinimum.Functions.RestrictedFunctions.count_user_posts_count_user_posts)
----------------------------------------------------------------------
```
-- https://github.com/Automattic/Co-Authors-Plus/actions/runs/5712043848/job/15474752693?pr=502

jeherve committed:
feat: surface linked guest author in Users list table

When a guest author is linked to a WordPress user, the user's post count
on the Users screen reflects the guest author's posts rather than the
user's own. That has long confused administrators because nothing on the
screen explains where those numbers come from. Add a new "Linked Guest
Author" column ahead of the post count, linking to the guest author's
edit screen, so the relationship is visible at a glance. Toggleable via
Screen Options like any other column.

Also harden the existing post-count rendering with sprintf and proper
escaping in passing.

GaryJones committed:
test: cover users list table column filters

Lock in the behaviour of the new "Linked Guest Author" column and the
existing post-count column so future regressions surface in CI rather
than during manual smoke tests on the Users screen. Covers column order
when the core Posts column is replaced, the no-op path when the Posts
column is already absent, both branches of the rendering filter (with
and without a linked guest author, with and without posts), and the
pass-through guarantee for columns the filter does not own.
